### PR TITLE
Adding tailwind css framework along with tooling

### PR DESCRIPTION
**Pull request by thirdknife**
#10 Adding tailwind css framework along with tooling, use Makefile targets to compile required tailwind components. Added air configuration to watch the directories to reload the application when the front end code changes.

image

### CODE REVIEW

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -1,5 +1,22 @@
 FROM mcr.microsoft.com/devcontainers/go:1-1.22-bookworm
 
+ENV NODE_VERSION 20.15.0
```
Comment (thirdknife) [inline .devcontainer/Dockerfile:3]: Adding instructions to install nvm and node

File: .air.toml [hunk 2]
```diff
@@ -41,4 +41,9 @@ tmp_dir = "tmp"
 
 [screen]
   clear_on_rebuild = false
-  keep_scroll = true
\ No newline at end of file
+  keep_scroll = true
+
+[proxy]
```
Comment (thirdknife) [inline .air.toml:46]: Proxy helps with air port to backend port and also hot reloads on fronted changes.

File: .air.toml [hunk 3]
```diff
@@ -13,7 +13,7 @@ tmp_dir = "tmp"
   exclude_unchanged = false
   follow_symlink = false
   full_bin = ""
-  include_dir = ["cmd", "views", "pkg"]
+  include_dir = ["cmd", "public", "views", "pkg"]
```
Comment (thirdknife) [inline .air.toml:16]: Added `public` folder for static content.

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,4 @@
+dev:
```
Comment (thirdknife) [inline Makefile:1]: Adding make targets.

File: cmd/main.go [hunk 5]
```diff
@@ -41,6 +41,7 @@ func newPage() Page {
 func main() {
 
 	e := echo.New()
+	e.Static("/public", "public")
```
Comment (thirdknife) [inline cmd/main.go:44]: Serve `/public` directory for public content.

File: go.mod [hunk 6]
```diff
@@ -2,17 +2,19 @@ module github.com/thirdknife/scoutingapp
 
 go 1.22.3
 
-require github.com/labstack/echo/v4 v4.12.0
+require (
```
Comment (thirdknife) [inline go.mod:5]: ran `go mod tidy`

File: package.json [hunk 7]
```diff
@@ -0,0 +1,5 @@
+{
+  "devDependencies": {
+    "tailwindcss": "^3.4.4"
```
Comment (thirdknife) [inline package.json:3]: only tailwind

File: tailwind.config.js [hunk 8]
```diff
@@ -0,0 +1,10 @@
+/** @type {import('tailwindcss').Config} */
+module.exports = {
+  mode: 'jit',
+  content: ['./views/**/*.html'],
```
Comment (thirdknife) [inline tailwind.config.js:4]: this makes sure any change in views will regenerate the tailwind final css.

File: cmd/main.go (hunk 5 above)
Comment (Omustardo) [reply cmd/main.go:44]: I don't do a lot of frontend, but maybe /static would be a more representative name? /public works too, especially if it's commonly used.

File: go.mod (hunk 6 above)
Comment (Omustardo) [reply go.mod:5]: This seems fine, but I'm not sure why it moved dependencies from indirect to direct. The golang changes in this branch shouldn't have required that. Anyway, probably fine.

**Review by Omustardo — APPROVED** [review]

File: cmd/main.go (hunk 5 above)
Comment (thirdknife) [reply cmd/main.go:44]: both are used, doesn't really matter.

File: go.mod (hunk 6 above)
Comment (thirdknife) [reply go.mod:5]: Yeah, it shouldn't have moved it to direct.

**Pull request merged**